atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1377,6 +1377,9 @@ private void commitWrites(TransactionService transactionService) {
             Timer.Context conflictsTimer = getTimer("commitCheckingForConflicts").time();
             throwIfConflictOnCommit(commitLocksToken, transactionService);
             long millisCheckingForConflicts = TimeUnit.NANOSECONDS.toMillis(conflictsTimer.stop());
+
+            sweepQueue.enqueue(writesByTable, getStartTimestamp());
```

Comment:
This may succeed, and then the `multiPut` might fail. Then when we read from the queue, we'll be trying to process writes that don't exist. Is this an issue?

Sanity check: `getStartTimestamp()` is a `Supplier<Long>`. How sure are we that calling it twice will return the same value, both now and in the future?

Reply:
That's by design, to avoid any case where we've written data but not updated the queue. If the `multiPut` fails (or only partially succeeds), we'll just issue some possibly empty range deletes. 

As an extra detail, we must have targeted sweep check if the transactions actually committed or not and clean up *only* the data they wrote if they did not commit. Otherwise, moving the location of the write would be a correctness issue - we may range delete valid data behind an uncommitted transaction.

The `getStartTimestamp()` issue is present all over this class. We could memoize locally to be defensive, but all normal suppliers that I know of are already memoized.

Reply:
Also, make sure to add tests for failed transactions :)

Reply:
Very good point. This will be done in a separate PR